pdns/filterpo.cc

```diff
@@ -214,7 +214,58 @@ bool DNSFilterEngine::getQueryPolicy(const DNSName& qname, const ComboAddress& c
         enabled = false;
```

Comment:
We only need the two-pass loop when we are matching a `DNSName` (to avoid computing all labels + wildcards as many times as the number of policies), you can mimic `bool DNSFilterEngine::getProcessingPolicy(const ComboAddress& address, const std::unordered_map<std::string,bool>& discardedPolicies, Policy& pol) const` here.